### PR TITLE
Heat 227

### PR DESCRIPTION
**Pull request by KalyanaVadlamani**

### CODE REVIEW

**Review by babuharidass11 — COMMENTED** [review]
squash all commits?

**Review by babuharidass11 — CHANGES_REQUESTED** [review]
updated

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -46,14 +47,31 @@ jobs:
...
+            echo "KUBE_CERT=${{ secrets.PRODUCTION_KUBE_CERT }}" >> $GITHUB_ENV
+            echo "KUBE_API=https://${{ secrets.PRODUCTION_KUBE_CLUSTER }}" >> $GITHUB_ENV
+          else
+            echo "KUBE_CLUSTER=${{ secrets.DEVELOPMENT_KUBE_CLUSTER }}" >> $GITHUB_ENV
+            echo "KUBE_NAMESPACE=${{ secrets.DEVELOPMENT_KUBE_NAMESPACE }}" >> $GITHUB_ENV
+            echo "KUBE_TOKEN=${{ secrets.DEVELOPMENT_KUBE_TOKEN }}" >> $GITHUB_ENV
+            echo "KUBE_CERT=${{ secrets.DEVELOPMENT_KUBE_CERT }}" >> $GITHUB_ENV
+            echo "KUBE_API=https://${{ secrets.DEVELOPMENT_KUBE_CLUSTER }}" >> $GITHUB_ENV
+          fi 
+
```
Comment (mattops) [inline .github/workflows/deploy.yml:66]: None of this section is needed, if you update the cloudplatform terraform code here, github.com/ministryofjustice/cloud-platform-terraform-serviceaccount?ref=1.0.0  - remove the environment string interpolation.

File: health_ping.py [hunk 2]
```diff
@@ -22,6 +22,19 @@
...
+# See strapi filter syntax https://docs.strapi.io/dev-docs/api/rest/filters-locale-publication
+# Example filter string = '&filters[name][$contains]=example'
+# sc_filter = '&filters[name][$contains]=book-a-prison-visit-staff-ui'
+sc_filter = os.getenv("sc_filter")
+sc_page_size = 10
+sc_pagination_page_size = f"&pagination[pageSize]={sc_page_size}"
+# Example Sort filter
+#SC_SORT='&sort=updatedAt:asc'
+sc_sort = ''
+sc_endpoint = f"{sc_api_endpoint}/v1/components?populate=environments{sc_filter}{sc_pagination_page_size}{sc_sort}"
```
Comment (mattops) [inline health_ping.py:35]: this is duplicated [line](https://github.com/ministryofjustice/hmpps-health-ping/pull/10/files#diff-73b875e568a6d799661f8bfb64f865de2790269e12df6410363ce2e9ed847a37L213)

File: health_ping.py [hunk 3]
```diff
@@ -22,6 +22,19 @@
 refresh_interval = int(os.getenv("REFRESH_INTERVAL","60"))
 log_level = os.environ.get('LOG_LEVEL', 'INFO').upper()
 
+# limit results for testing/dev
+# See strapi filter syntax https://docs.strapi.io/dev-docs/api/rest/filters-locale-publication
+# Example filter string = '&filters[name][$contains]=example'
+# sc_filter = '&filters[name][$contains]=book-a-prison-visit-staff-ui'
+sc_filter = os.getenv("sc_filter")
```
Comment (mattops) [inline health_ping.py:29]: see comment above, not this is correct?

File: health_ping.py [hunk 4]
```diff
@@ -22,6 +22,19 @@
...
+sc_page_size = 10
+sc_pagination_page_size = f"&pagination[pageSize]={sc_page_size}"
```
Comment (mattops) [inline health_ping.py:31]: the pagination variables will not work with out other code to iterate over the pages. Suggest you remove this block that has been copy/pasted.

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -46,14 +47,31 @@ jobs:
...
         with:
-          environment: ${{ inputs.environment }}
-          version: ${{ inputs.version }}
-          api: https://${{ secrets.DEVELOPMENT_KUBE_CLUSTER }}
-          cert: ${{ secrets.DEVELOPMENT_KUBE_CERT }}
-          cluster: ${{ secrets.DEVELOPMENT_KUBE_CLUSTER }}
-          namespace: ${{ secrets.DEVELOPMENT_KUBE_NAMESPACE }}
-          token: ${{ secrets.DEVELOPMENT_KUBE_TOKEN }}
+          environment: ${{ inputs.github_environment }}
+          version: ${{ inputs.github_environment }}
```
Comment (mattops) [inline .github/workflows/deploy.yml:71]: Why is version being set to `github_environment` ?

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -31,10 +31,11 @@ on:
...
           - dev
-      version:
-        description: Image version
-        type: string
-        required: true
+          - prod
+      # version:
+      #   description: Image version
+      #   type: string
+      #   required: true
```
Comment (mattops) [inline .github/workflows/deploy.yml:38]: Why is this commented out?

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (KalyanaVadlamani) [reply .github/workflows/deploy.yml]: updated, (none of these were part of the original PR, I had some commits and lost some during squash)

File: .github/workflows/deploy.yml (hunk 6 above)
Comment (KalyanaVadlamani) [reply .github/workflows/deploy.yml]: uncommented now. (lost them during my squash)

File: health_ping.py [hunk 7]
```diff
@@ -22,6 +21,18 @@
...
 
+# limit results for testing/dev
+# See strapi filter syntax https://docs.strapi.io/dev-docs/api/rest/filters-locale-publication
+# Example filter string = '&filters[name][$contains]=example'
+# sc_api_filter = '&filters[name][$contains]=book-a-prison-visit-staff-ui'
+sc_api_filter = os.getenv("SERVICE_CATALOGUE_FILTER", '')
+
+# Example Sort filter
+#SC_SORT='&sort=updatedAt:asc'
+sc_sort = ''
```
Comment (mattops) [inline health_ping.py:32]: this variable is not used.

File: .github/workflows/pipeline.yml [hunk 8]
```diff
@@ -5,21 +5,43 @@ on:
...
+  deploy_to_prod:
+    name: Deploy to prod
+    uses: ./.github/workflows/deploy.yml
+    needs:
+      - deploy_to_dev # wait for the deploy_to_dev job to complete
+    with:
+      github_environment: production
+      environment: prod
+      version: ${{ github.event.inputs.version }}
+    if: github.event_name != 'schedule' # Skip this job if the workflow is triggered by schedule
```
Comment (mattops) [inline .github/workflows/pipeline.yml:46]: Are we ever going to run this on a schedule?

File: health_ping.py (hunk 7 above)
Comment (KalyanaVadlamani) [reply health_ping.py]: rectified now

File: .github/workflows/pipeline.yml (hunk 8 above)
Comment (KalyanaVadlamani) [reply .github/workflows/pipeline.yml]: Removing now, however can be used if at all we find there is a need for regular updates, scheduled-maintenance etc

**Review by mattops — APPROVED** [review]

**Review by babuharidass11 — APPROVED** [review]
LGTM

**Pull request merged**